webhooks.go

```diff
@@ -0,0 +1,231 @@
+package resend
+
+import (
+	"context"
+	"net/http"
+)
+
+// CreateWebhookRequest represents the parameters for creating a webhook
+type CreateWebhookRequest struct {
+	Endpoint string   `json:"endpoint"`
+	Events   []string `json:"events"`
+}
+
+// CreateWebhookResponse represents the response from creating a webhook
+type CreateWebhookResponse struct {
+	Object        string `json:"object"`
+	Id            string `json:"id"`
+	SigningSecret string `json:"signing_secret"`
+}
+
+// Webhook represents a webhook object
+type Webhook struct {
+	Object        string   `json:"object"`
+	Id            string   `json:"id"`
+	CreatedAt     string   `json:"created_at,omitempty"`
+	Status        string   `json:"status,omitempty"`
+	Endpoint      string   `json:"endpoint,omitempty"`
+	Events        []string `json:"events,omitempty"`
```

Comment:
I wonder if we should make this a typed constant as in:

```
const (
	EventEmailSent      WebhookEvent = "email.sent"
	EventEmailDelivered WebhookEvent = "email.delivered"
	EventEmailBounced   WebhookEvent = "email.bounced"
	EventEmailOpened    WebhookEvent = "email.opened"
	EventEmailClicked   WebhookEvent = "email.clicked"
    // more events
)
```

The problem with this is that (IIRC) we then can't call the endpoint with just a string:

```
SomeFunctionThatTakesEvent(EventEmailSent)     // OK
SomeFunctionThatTakesEvent("email.unknown")    // compile-time error
```

What do you think? I think a raw string is fine, but just double checking if we want an "enum" kind of behavior here.